### PR TITLE
Cleanup phase II

### PR DESCRIPTION
**Pull request by kuldeepaggarwal**
```
- seperate shared examples with specs
- remove code which was causing `WARNING`s
- add `FIXME`s and some code optimization
```

@kristianmandrup @ankush86: Please review.

### CODE REVIEW

File: lib/cms/models/block.rb [hunk 1]
```diff
@@ -21,6 +21,8 @@ class Block < Cms::Models::Document
       localize_fields :title, :description
       blueprint!
 
+      # [FIXME]: We are already tracking +:all+ in +ContentItem+ module, do
+      # we really this?
       tracks :name
```
Comment (kuldeepaggarwal) [inline lib/cms/models/block.rb:26]: @kristianmandrup @ankush86 Please respond on this.

File: lib/cms/models/image.rb [hunk 2]
```diff
@@ -25,6 +25,8 @@ class Image < Cms::Models::Document
       fields_validation
       localize_fields :title, :description
 
+      # [FIXME]: We are already tracking +:all+ in +ContentItem+ module, do
+      # we really this?
       tracks :name
```
Comment (kuldeepaggarwal) [inline lib/cms/models/image.rb:30]: @kristianmandrup @ankush86 Please respond on this.

Comment (kristianmandrup) [inline lib/cms/models/image.rb:30]: Yes, you can remove `tracks :name`

File: lib/cms/models/block.rb (hunk 1 above)
Comment (kristianmandrup) [inline lib/cms/models/block.rb:26]: yes, remove

**Pull request merged**